### PR TITLE
Refactor permission check logic, add WebView permission UI

### LINKED ISSUE #3207: Cannot use SIP client, as there is no way to grant microphone access to OpenHAB-Android
I create a SIP-client card in Main UI 3.4. With OpenHAB-Android 3.0.5-beta I load the main page.  OpenHAB-Android shows:

> openHAB
> To use the SIP widget you must allow microphone/camera access in your browser and reload this page.

I do not know how to allow microphone/camera access to OpenHAB.

In F-Droid the application does not show that it requires Camera/Microphone access.

### PR DESCRIPTION
**Pull request by maniac103**
Fixes #3207

### CODE REVIEW

**Comment by maniac103** [comment]
This makes Main UI not ask for permissions anymore, but I can't check whether it actually works. Testing by anybody with a SIP account would be appreciated.

**Comment by mueller-ma** [comment]
Can you upload a debug apk, so @dilyanpalauzov can test it?

**Comment by dilyanpalauzov** [comment]
Test account: u: aaa secret E0abc server wss://sip.bapha.be/ws/ , the numbers from http://enum-test.at/index_en.html can be dialed (without spaces).

**Comment by maniac103** [comment]
Test apk (rename to .apk): 
[mobile-foss-beta-release.zip](https://github.com/openhab/openhab-android/files/10720069/mobile-foss-beta-release.zip)

**Comment by dilyanpalauzov** [comment]
When I connect over insecure HTTP openhab says: 

> **openHAB**
> Please ensure that HTTPS is in use and WebRTC is supported in this browser.

Over secure HTTPS:

The bottom shows a message for short time

> This page requires additional permissions to work properly [PEMRIT]

If I do not click it fast enough, the message disappears and another message appears:

> **openHAB**
> To use the SIP widget you must allow microphoe/camera access in your browser and reload this page.
[OK]

When I call this message is sent:
```
INVITE sip:v1@example.org SIP/2.0
Via: SIP/2.0/WSS rrioh3atelui.invalid;branch=z9hG4bK6085320
Max-Forwards: 69
To: <sip:v1@example.org>
From: <sip:openhab-vn@example.org>;tag=vkjtkfgfkq
Call-ID: fip5cpc2vjrpa5a14k7u
CSeq: 5881 INVITE
Contact: <sip:openhab-vn@example.org;gr=urn:uuid:21dc2176-92da-48db-a2f2-11d6d26c2d12>
Content-Type: application/sdp
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: gruu,ice,replaces,outbound
User-Agent: JsSIP 3.9.1
Content-Length: 1486

v=0
o=- 6290557329234886134 2 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE 0
a=extmap-allow-mixed
a=msid-semantic: WMS 2yHr2ToLglVrMwJ45LgH84IaPgwuldyiARGC
m=audio 38306 UDP/TLS/RTP/SAVPF 111 63 103 9 0 8 105 13 110 113 126
c=IN IP4 192.168.0.146
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:244150863 1 udp 2113937151 192.168.0.146 38306 typ host generation 0 network-cost 999
a=ice-ufrag:Fqfv
=ice-pwd:S4jIPBPXaiHzmpkAYMHf7xtd
a=ice-options:trickle
a=fingerprint:sha-256 4B:20:C5:E1:7B:E7:A8:DB:A2:41:3F:DD:54:96:EA:71:CE:CA:11:98:45:A3:C4:0C:D5:5A:55:71:BF:F5:B6:20
a=setup:actpass
a=mid:0
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:4 urn:ietf:params:rtp-hdrext:sdes:mid
a=sendrecv
a=msid:2yHr2ToLglVrMwJ45LgH84IaPgwuldyiARGC 9f33140d-2900-43df-b948-2c1e45d44b44
a=rtcp-mux
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:63 red/48000/2
a=fmtp:63 111/111
a=rtpmap:103 ISAC/16000
a=rtpmap:9 G722/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:105 CN/16000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
a=ssrc:2641115819 cname:Ggghwn4E2U0JYTHD
a=ssrc:2641115819 msid:2yHr2ToLglVrMwJ45LgH84IaPgwuldyiARGC 9f33140d-2900-43df-b948-2c1e45d44b44
```

The telephone rings, but no connection is established - no voice is transferred, even if the line is open for short time.  When I call using OpenHAB from Firefox/Android this messages is sent
```
INVITE sip:v1@example.org SIP/2.0
Via: SIP/2.0/WSS h2mk02845dvb.invalid;branch=z9hG4bK4181718
Max-Forwards: 69
To: <sip:v1@bapha.be>
From: <sip:openhab-vn@example.org>;tag=qa9kg4fn8l
Call-ID: 4ngipslmbad1smtt6e68
CSeq: 1166 INVITE
Contact: <sip:openhab-vn@example.org;gr=urn:uuid:f6d930b6-f68c-45e4-ba17-89513c0a25ca>
Content-Type: application/sdp
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: gruu,ice,replaces,outbound
User-Agent: JsSIP 3.9.1
Content-Length: 1316

v=0
o=mozilla...THIS_IS_SDPARTA-99.0 9086774203835705029 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 58:79:30:25:47:0D:87:32:3C:68:65:29:2F:96:EA:86:3B:EC:98:86:54:FD:AA:AC:50:53:7F:96:8A:E9:94:09
a=group:BUNDLE 0
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 34947 UDP/TLS/RTP/SAVPF 109 9 0 8 101
c=IN IP4 192.168.0.146
a=candidate:0 1 UDP 2122252543 192.168.0.146 34947 typ host
a=candidate:1 1 TCP 2105524479 192.168.0.146 9 typ host tcptype active
a=candidate:0 2 UDP 2122252542 192.168.0.146 48590 typ host
a=candidate:1 2 TCP 2105524478 192.168.0.146 9 typ host tcptype active
a=sendrecv
a=end-of-candidates
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:9eaa79d1d9a767f573e68ef56cfbd447
a=ice-ufrag:01f621d4
a=mid:0
a=msid:{13f5bead-665b-42b7-99ac-2c63a393e1f0} {52e3867e-2544-489d-be4f-5361d00ae2dd}
a=rtcp:48590 IN IP4 192.168.0.146
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:2286785542 cname:{ff550288-b10d-4fee-87dc-5ad5284dedea}
```
and connection is established (voice is exchanged).

* It shall work also over insecure HTTP
* The message `This page requires additional permissions to work properly [PEMRIT]` shall not disappear without user inetraction.  In particular without user interaction the message shall not be replaced with a new one stating insufficient prvileges.
* Any idea how INVITEs differ significantly, so that one works, and the other not?

**Comment by maniac103** [comment]
> It shall work also over insecure HTTP

That's a Main UI limitation, we can not do anything about this here.

> The message This page requires additional permissions to work properly [PEMRIT] shall not disappear without user inetraction

That can be done, but I'm not sure why we should treat the web page permissions differently from other app permission requests; currently the duration is the same for both (`Snackbar.LENGTH_LONG`). @mueller-ma Thoughts on this?

> Any idea how INVITEs differ significantly, so that one works, and the other not?

Is there anything relevant (in particular `chromium` lines) in the app log? On first sight it feels like a codec issue, because in the app INVITE log there's more codecs than in the Firefox one. What about Chrome, how does it look there?

**Comment by dilyanpalauzov** [comment]
>>   The message This page requires additional permissions to work properly [PERMIT] shall not disappear without user inetraction

> That can be done, but I'm not sure why we should treat the web page permissions differently from other app permission requests; currently the duration is the same for both (Snackbar.LENGTH_LONG). @mueller-ma Thoughts on this?

I perceive this as disappearing too fast.  There is not enough time to read all the displayed information and this tooltip on the botton.  One can just see that a tooltip disappears, so the user has to go back,, reload the page and intentionally focus from the beginning on reading the tooltip, before it disappears.

I cannot say right now how it works with Chrome.

I have created a second account for experiments.  User bbb Pw: A0abc , domain bapha.be .  These two tests accounts can be used within OpenHAB or other SIP-clients.  It is also possible to call via https://sip.bapha.be/ by typing aaa or bbb .  I think it is better if you can try the changes yourself, including intercom between two OpenHAB-Android instances.

There might be problems with the SIP/RTP server from time to time, it is currently in test phase.

**Comment by maniac103** [comment]
>  I think it is better if you can try the changes yourself, including intercom between two OpenHAB-Android instances.

The thing is: I already tried that, but am getting this:
```
oh-sipclient: Calling +437200101011
oh-sipclient: Call failed. Reason: Rejected
```

This applies to trying both in Firefox on desktop as well as the Android app.

**Comment by dilyanpalauzov** [comment]
About +437200101011 it is called over SIP enum and the final server does not support encrypted traffic.  So the intermediate server (doing the ENUM redirection) replaces in theroy SRTP with RTP.  When calling from https://sip.bapha.be/ just entering the above number, the call works, when started from Firefox, does not work when started from Chromium.  Under Epiphany(WebKit) sip.bapha.be cannot be loaded.  I will investigate why the call does not work under Chromium, but it will take some time.  I agree testing with +437200101011 is not very reliable for the above reasons.

However for the other two accounts the SIP-server is not supposed to modify the encryption data in the SDP message, so both provided test accounts shall be suitable for testing.

**Comment by maniac103** [comment]
Well, I tried this:

```
config:
  label: Overview
blocks: []
masonry:
  - component: oh-masonry
    slots:
      default:
        - component: oh-sipclient-card
          config:
            websocketUrl: wss://sip.bapha.be/ws/
            username: aaa
            password: E0abc
            phonebook: bbb
            domain: example.org
grid: null
canvas: null
```

Is this correct? If so, it doesn't work:

```
REGISTER sip:example.org SIP/2.0
Via: SIP/2.0/WSS 2l0obnd2scvl.invalid;branch=z9hG4bK8825045
Max-Forwards: 69
To: <sip:aaa@example.org>
From: <sip:aaa@example.org>;tag=sebduc1g25
Call-ID: lp5hleiafi7hbt61ed2ahk
CSeq: 2 REGISTER
Authorization: Digest algorithm=MD5, username="aaa", realm="example.org", nonce="Y+of12PqHqtMQE/bZ2c/1MEVDxX3GcPGIaWc/8RR57HaBx9Yhgv4ug==", uri="sip:example.org", response="df7fa23ab5fd597cddbb807522b24e26"
Contact: <sip:e80srn5n@2l0obnd2scvl.invalid;transport=ws>;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:2d777b47-2788-4bdd-809b-2cd9fbaee9c0>";expires=600
Expires: 600
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: path,gruu,outbound
User-Agent: JsSIP 3.9.1
Content-Length: 0

SIP/2.0 401 Unauthorized
Via: SIP/2.0/WSS 2l0obnd2scvl.invalid;branch=z9hG4bK8825045;rport=45266;received=144.76.142.78
To: <sip:aaa@example.org>;tag=44af9dbde0a62ae84f5c1ff272adb901.6c320000
From: <sip:aaa@example.org>;tag=sebduc1g25
Call-ID: lp5hleiafi7hbt61ed2ahk
CSeq: 2 REGISTER
WWW-Authenticate: Digest realm="example.org", nonce="Y+of12PqHqtMQE/bZ2c/1MEVDxX3GcPGIaWc/8RR57HaBx9Yhgv4ug=="
Content-Length: 0

INVITE sip:bbb@example.org SIP/2.0
Via: SIP/2.0/WSS 2l0obnd2scvl.invalid;branch=z9hG4bK5462830
Max-Forwards: 69
To: <sip:bbb@example.org>
From: <sip:aaa@example.org>;tag=19152h80t7
Call-ID: svpeaai7d496f17pcvdk
CSeq: 2448 INVITE
Contact: <sip:e80srn5n@2l0obnd2scvl.invalid;transport=ws;ob>
Content-Type: application/sdp
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: ice,replaces,outbound
User-Agent: JsSIP 3.9.1
Content-Length: 2348

v=0
o=mozilla...THIS_IS_SDPARTA-99.0 8356329145047190017 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 3B:33:75:42:19:31:2E:87:46:EB:D5:12:99:9C:FF:03:DA:87:63:5C:81:9F:87:97:70:6C:60:09:9A:EB:8F:06
a=group:BUNDLE 0
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 47394 UDP/TLS/RTP/SAVPF 109 9 0 8 101
c=IN IP4 192.168.2.26
a=candidate:0 1 UDP 2122055935 192.168.2.26 47394 typ host
a=candidate:1 1 UDP 2122187007 fd07:2fa:6cff:2402::2d0 46012 typ host
a=candidate:2 1 UDP 2122252543 2001:1438:4010:702:7f85:bf3b:608f:df07 60475 typ host
a=candidate:3 1 UDP 2122121471 fd07:2fa:6cff:2402:df29:b88d:caa6:258f 35660 typ host
a=candidate:4 1 TCP 2105327871 192.168.2.26 9 typ host tcptype active
a=candidate:5 1 TCP 2105458943 fd07:2fa:6cff:2402::2d0 9 typ host tcptype active
a=candidate:6 1 TCP 2105524479 2001:1438:4010:702:7f85:bf3b:608f:df07 9 typ host tcptype active
a=candidate:7 1 TCP 2105393407 fd07:2fa:6cff:2402:df29:b88d:caa6:258f 9 typ host tcptype active
a=candidate:0 2 UDP 2122055934 192.168.2.26 52466 typ host
a=candidate:1 2 UDP 2122187006 fd07:2fa:6cff:2402::2d0 55618 typ host
a=candidate:2 2 UDP 2122252542 2001:1438:4010:702:7f85:bf3b:608f:df07 39360 typ host
a=candidate:3 2 UDP 2122121470 fd07:2fa:6cff:2402:df29:b88d:caa6:258f 59075 typ host
a=candidate:4 2 TCP 2105327870 192.168.2.26 9 typ host tcptype active
a=candidate:5 2 TCP 2105458942 fd07:2fa:6cff:2402::2d0 9 typ host tcptype active
a=candidate:6 2 TCP 2105524478 2001:1438:4010:702:7f85:bf3b:608f:df07 9 typ host tcptype active
a=candidate:7 2 TCP 2105393406 fd07:2fa:6cff:2402:df29:b88d:caa6:258f 9 typ host tcptype active
a=sendrecv
a=end-of-candidates
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:0a527cf9ac152e5d3781d03b3288d877
a=ice-ufrag:d45c6754
a=mid:0
a=msid:{5ba4946f-59a4-4f10-aa58-6d0a2229b0bb} {fa0f7f11-4b63-4977-9d13-824c79707f1a}
a=rtcp:52466 IN IP4 192.168.2.26
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:1882316382 cname:{4dd92e30-13ca-4541-855f-d7be98688304}

SIP/2.0 403 Not relaying
Via: SIP/2.0/WSS 2l0obnd2scvl.invalid;branch=z9hG4bK5462830;rport=45266;received=144.76.142.78
To: <sip:bbb@example.org>;tag=44af9dbde0a62ae84f5c1ff272adb901.6c320000
From: <sip:aaa@example.org>;tag=19152h80t7
Call-ID: svpeaai7d496f17pcvdk
CSeq: 2448 INVITE
Content-Length: 0
```

(All of this is with Firefox on desktop, FWIW)

**Comment by dilyanpalauzov** [comment]
> domain: example.org

The domain is bapha.be, it shall be used also in WW-Authenticate/Authorization headers, SIP Request-URI, To: and From: headers.

**Comment by maniac103** [comment]
Okay, now authentication is successful, but I get a 404:

```
INVITE sip:bbb@bapha.be SIP/2.0
Via: SIP/2.0/WSS 392vnta1drp5.invalid;branch=z9hG4bK9517281
Max-Forwards: 69
To: <sip:bbb@bapha.be>
From: <sip:aaa@bapha.be>;tag=ehnik6ak6r
Call-ID: 3lmfkb5ktvou1chlk91c
CSeq: 2348 INVITE
Proxy-Authorization: Digest algorithm=MD5, username="aaa", realm="bapha.be", nonce="Y+onYWPqJjXb9eOgqplKDG+/H1HalEFikSHA6wu3mldxShsiv09Tvg==", uri="sip:bbb@bapha.be", response="02e96ce3ba977ca61ec80eebba6b12bf"
Contact: <sip:aaa@bapha.be;gr=urn:uuid:2fdcbca2-70a8-4968-a4c8-762ce8e874a6>
Content-Type: application/sdp
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO,NOTIFY
Supported: gruu,ice,replaces,outbound
User-Agent: JsSIP 3.9.1
Content-Length: 2348

v=0
o=mozilla...THIS_IS_SDPARTA-99.0 5904181439673022480 0 IN IP4 0.0.0.0
s=-
t=0 0
a=sendrecv
a=fingerprint:sha-256 AF:46:B3:BD:69:FA:F4:E8:99:4C:C2:0E:4A:8C:E1:1A:23:45:79:9C:91:D4:1D:F0:98:3D:26:B9:0E:29:08:12
a=group:BUNDLE 0
a=ice-options:trickle
a=msid-semantic:WMS *
m=audio 37214 UDP/TLS/RTP/SAVPF 109 9 0 8 101
[...snip...]
a=sendrecv
a=end-of-candidates
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2/recvonly urn:ietf:params:rtp-hdrext:csrc-audio-level
a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
a=fmtp:109 maxplaybackrate=48000;stereo=1;useinbandfec=1
a=fmtp:101 0-15
a=ice-pwd:a5f453f8554d9e07232a86677cca7c21
a=ice-ufrag:25ad387b
a=mid:0
a=msid:{6c8ebc69-8413-440b-a665-d17b6e88414b} {d7d94e56-1470-4a50-905d-9b95bfc9c65f}
a=rtcp:34865 IN IP4 192.168.2.26
a=rtcp-mux
a=rtpmap:109 opus/48000/2
a=rtpmap:9 G722/8000/1
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:101 telephone-event/8000
a=setup:actpass
a=ssrc:1836952315 cname:{4ceeee23-f731-4bb1-843a-7e782d6368a7}

SIP/2.0 404 Not Found
Via: SIP/2.0/WSS 392vnta1drp5.invalid;branch=z9hG4bK9517281;rport=53722;received=144.76.142.78
To: <sip:bbb@bapha.be>;tag=f0c1e5b00382184fd85ea227e9804e16-cef30000
From: <sip:aaa@bapha.be>;tag=ehnik6ak6r
Call-ID: 3lmfkb5ktvou1chlk91c
CSeq: 2348 INVITE
Authentication-Info: nextnonce="Y+onYWPqJjXb9eOgqplKDG+/H1HalEFikSHA6wu3mldxShsiv09Tvg==", qop=, rspauth="c58c10f4ec42b29120b1bcc8431e9231", cnonce="", nc=
Content-Length: 0
```

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/fragments/SendDeviceInfoSettingsFragment.kt [hunk 1]
```diff
@@ -116,60 +102,33 @@ class SendDeviceInfoSettingsFragment :
...
-                        R.string.settings_phone_state_permission_denied
-                    )
-                    phoneStatePref.setValue(checked = false)
-                } else {
-                    BackgroundTasksManager.scheduleWorker(context, PrefKeys.SEND_PHONE_STATE, true)
-                }
+    private fun getAndInitPreferenceForPermissionRequest(
+        prefKey: String,
+        permDeniedSnackBarTag: String,
+        @StringRes permDeniedSnackBarTextResId: Int
```
Comment (mueller-ma) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/fragments/SendDeviceInfoSettingsFragment.kt:108]: In the Android framework the `b` of snackbar isn't a capital letter.

**Comment by dilyanpalauzov** [comment]
I have provided two accounts for testing: aaa and bbb.  Aaa can only call bbb, after bbb has registered (has issued REGISTER) and has not disconnected.  Otherwise aaa cannot call bbb .  The same is valid in the opposite direction.

At the same time, if only aaa has registered it can be called from https://sip.bapha.be by entering aaa .  Using Firefox vs Chrome when loading https://sip.bapha.be might lead to different results.

Without looking is the details I assume that bbb has not registered and aaa tried to call bbbb.

**Comment by maniac103** [comment]
> Without looking in the details I assume that bbb has not registered and aaa tried to call bbb.

That's indeed the case, thanks for the explanation. One might have noticed already I have no prior SIP knowledge ;-)

Now I'm getting this in the log:
```
02-13 13:24:26.309 22093 22323 E chromium: [ERROR:sdp_offer_answer.cc(750)] Failed to set remote offer sdp: SDES and DTLS-SRTP cannot be enabled at the same time.
02-13 13:24:26.313 22000 22000 I chromium: [INFO:CONSOLE(7)] "oh-sipclient: Call failed. Reason: WebRTC Error"
```

FWIW, I get the same thing in Chrome browser as well, so there seems to be a misconfiguration and/or incompatibility with Chrome here.

**Comment by maniac103** [comment]
> I think a snackbar at start can be missed and missing permissions may cause larger issues, e.g. notifications won't work at all. You can increase the duration to Snackbar.LENGTH_INDEFINITE.

Done.

**Comment by dilyanpalauzov** [comment]
There are these cases possible:

A) aaa registers in OpenHAB on Firefox, aaa is called from https://sip.bapha.be from Chrome/Chromium

B) aaa registers in OpenHAB on Firefox, aaa is called from https://sip.bapha.be from Firefox

C) aaa registers in OpenHAB on Chrome/Chromium, aaa is called from https://sip.bapha.be from Firefox

D) aaa registers in OpenHAB on Chrome/Chromium, aaa is called from https://sip.bapha.be from Chrome/Chromium

E) aaa registers in OpenHAB-Android, aaa is called from https://sip.bapha.be from Chrome/Chromium

F) aaa registers in OpenHAB-Android, aaa is called from https://sip.bapha.be from Firefox

G) aaa registers in OpenHAB on Firefox, bbb registers in Openhab on Chrome/Chromium.  aaa calls bbb.

H) aaa registers in OpenHAB on Firefox, bbb registers in Openhab on Firefox.  aaa calls bbb.

I) aaa registers in OpenHAB on Firefox, bbb registers in Openhab-Android.  aaa calls bbb.

J) aaa registers in OpenHAB on Chrome/Chromium, bbb registers in Openhab on Chrome/Chromium.  aaa calls bbb.

K) aaa registers in OpenHAB on Chrome/Chromium, bbb registers in Openhab on Firefox.  aaa calls bbb.

L) aaa registers in OpenHAB on Chrome/Chromium, bbb registers in Openhab-Android.  aaa calls bbb.

M) aaa registers in OpenHAB-Android, bbb registers in Openhab on Chrome/Chromium.  aaa calls bbb.

N) aaa registers in OpenHAB-Android, bbb registers in Openhab on Firefox.  aaa calls bbb.

O) aaa registers in OpenHAB-Android, bbb registers in Openhab-Android.  aaa calls bbb.

Is somebody willing to try all the cases?

Which of the above cases produces:

> 02-13 13:24:26.309 22093 22323 E chromium: [ERROR:sdp_offer_answer.cc(750)] Failed to set remote offer sdp: SDES and DTLS-SRTP cannot be enabled at the same time.
> 02-13 13:24:26.313 22000 22000 I chromium: [INFO:CONSOLE(7)] "oh-sipclient: Call failed. Reason: WebRTC Error"

Can you copy the request which produces server error and the response (the error)?

A server error can happen if more than 10 REGISTRATIONs for the same user have to be maintained by the server. at the same time  Each REGISTRATION is kept 600s (Expire: 600 header) in memory.

I believe this changeset does improve the situation, so once a new beta is released with the changes here, more testing can be performed.

**Comment by mueller-ma** [comment]
IMO it's getting a bit off-topic here. It's probably a good idea to forward permission requests from WebViews to the OS, so I will review this part.
When the SIP client isn't working then, it's probably an issue with the WebView. Maybe you can update the following package: https://play.google.com/store/apps/details?id=com.google.android.webview

**Comment by dilyanpalauzov** [comment]
> IMO it's getting a bit off-topic here. It's probably a good idea to forward permission requests from WebViews to the OS, so I will review this part. When the SIP client isn't working then, it's probably an issue with the WebView. Maybe you can update the following package: https://play.google.com/store/apps/details?id=com.google.android.webview

As far as I am concerned, I have the latest WebKit.  Even if the voice does not work, this is not a problem of this ticket: it could be WebKit, jssip-related, the voip server does not perform transcoding… 

I would like to see a release, where the user is permitted to grant the microphone and camera permission to openHAB-android.  Other problems can be tackled in different tickets.

**Review by mueller-ma — APPROVED** [review]

**Pull request merged**